tests/integration/book.test.js

```diff
@@ -0,0 +1,177 @@
+const { expect } = require("chai");
+const request = require("supertest");
+const { Book } = require("../../src/models");
+const app = require("../../src/app");
+const dataFactory = require("../helpers/dataFactory");
+
+describe("/books", () => {
+  before(async () => Book.sequelize.sync());
+
+  beforeEach(async () => {
+    await Book.destroy({ where: {} });
+  });
+
+  describe("with no books in the database", () => {
+    describe("POST", () => {
+      it("creates a book", async () => {
+        const getBookData = dataFactory.bookData();
+
+        const response = await request(app).post("/books").send({
+          title: getBookData.title,
+          isbn: getBookData.isbn,
+        });
+
+        expect(response.status).to.equal(201);
+        expect(response.body.title).to.equal(getBookData.title);
+        expect(response.body.isbn).to.equal(getBookData.isbn);
+
+        const bookDocument = await Book.findByPk(response.body.id, {
+          raw: true,
+        });
+
+        expect(bookDocument.title).to.equal(getBookData.title);
+        expect(bookDocument.isbn).to.equal(getBookData.isbn);
+      });
+
+      it("returns an error if title is empty", async () => {
+        const bookData = dataFactory.bookData();
+
+        const response = await request(app).post("/books").send({
+          title: "",
+          isbn: bookData.isbn,
+        });
+
+        expect(response.status).to.equal(400);
+        expect(response.body).to.equal(`"title" is not allowed to be empty`);
+      });
+    });
+  });
+
+  describe("with records in the database", () => {
+    let books;
+
+    beforeEach(async () => {
+      const bookExamples = [];
+
+      for (let i = 0; i < 3; i++) {
+        bookExamples.push(dataFactory.bookData());
+      }
```

Comment:
This test is failing due to duplicates of the isbn generated by your faker function. You can get around this by ensuring the values are unique by doing something like this:

```
      const bookExamples = [];
      const existingIsbns = [];
      while (bookExamples.length < 3) {
        const newBook = dataFactory.bookData();
        if (!existingIsbns.includes(newBook.isbn)) {
          bookExamples.push(newBook);
          existingIsbns.push(newBook.isbn)
        }
      }
```